cmd/kube-apiserver/app/options/options_test.go

```diff
@@ -28,15 +28,217 @@ func TestAddFlagsFlag(t *testing.T) {
+	}
+	if s.Authorization.Mode != "AlwaysAllow" {
+		t.Errorf("Expected s.Authorization.Mode to be AlwaysAllow by default")
+	}
+	if s.Authorization.PolicyFile != "" {
+		t.Errorf("Expected s.Authorization.PolicyFile to be empty by default")
+	}
+	if s.Authorization.WebhookCacheAuthorizedTTL != 300000000000 {
+		t.Errorf("Expected s.Authorization.WebhookCacheAuthorizedTTL to be 300000000000 by default")
+	}
+	if s.Authorization.WebhookCacheUnauthorizedTTL != 30000000000 {
+		t.Errorf("Expected s.Authorization.WebhookCacheUnauthorizedTTL to be 30000000000 by default")
+	}
+	if s.Authorization.WebhookConfigFile != "" {
+		t.Errorf("Expected s.Authorization.WebhookConfigFile to be empty by default")
+	}
+	if s.SecureServing.BindAddress.String() != "0.0.0.0" {
+		t.Errorf("Expected s.SecureServing.BindAddress to be 0.0.0.0 by default")
+	}
+	if s.Authentication.ClientCert.ClientCA != "" {
+		t.Errorf("Expected s.Authentication.ClientCert.ClientCA to be empty by default")
+	}
+	if s.CloudProvider.CloudConfigFile != "" {
+		t.Errorf("Expected s.CloudProvider.CloudConfigFile to be empty by default")
+	}
+	if s.CloudProvider.CloudProvider != "" {
+		t.Errorf("Expected s.CloudProvider.CloudProvider to be empty by default")
+	}
+	if s.GenericServerRunOptions.CorsAllowedOriginList != nil {
+		t.Errorf("Expected s.GenericServerRunOptions.CorsAllowedOriginList to be empty by default")
+	}
+	if s.EnableAggregatorRouting {
+		t.Errorf("Expected s.EnableAggregatorRouting to be false by default")
+	}
+	if !s.EnableLogsHandler {
+		t.Errorf("Expected s.EnableLogsHandler to be true by default")
+	}
 	if s.Features.EnableSwaggerUI {
-		t.Errorf("Expected s.EnableSwaggerUI to be false by default")
+		t.Errorf("Expected s.Features.EnableSwaggerUI to be false by default")
 	}
 
 	args := []string{
+		"--advertise-address=192.168.10.10",
+		"--allow-privileged=false",
+		"--anonymous-auth=false",
+		"--apiserver-count=5",
+		"--audit-log-maxage=11",
+		"--audit-log-maxbackup=12",
+		"--audit-log-maxsize=13",
+		"--audit-log-path=/var/log",
+		"--audit-policy-file=/policy",
+		"--audit-webhook-config-file=/webhook-config",
+		"--audit-webhook-mode=blocking",
+		"--authentication-token-webhook-cache-ttl=3m",
+		"--authentication-token-webhook-config-file=/token-webhook-config",
+		"--authorization-mode=AlwaysDeny",
+		"--authorization-policy-file=/policy",
+		"--authorization-webhook-cache-authorized-ttl=3m",
+		"--authorization-webhook-cache-unauthorized-ttl=1m",
+		"--authorization-webhook-config-file=/webhook-config",
+		"--bind-address=192.168.10.20",
+		"--client-ca-file=/client-ca",
+		"--cloud-config=/cloud-config",
+		"--cloud-provider=daocloud",
+		"--cors-allowed-origins=10.10.10.100,10.10.10.200",
+		"--enable-aggregator-routing=true",
+		"--enable-logs-handler=false",
 		"--enable-swagger-ui=true",
 	}
 	f.Parse(args)
+
+	// Check option --advertise-address
+	if s.GenericServerRunOptions.AdvertiseAddress.String() != "192.168.10.10" {
```

Comment:
How about replacing all the single if-then clauses with a DeepEqual and a big struct? Would be much easier to  maintain.